kolibri/plugins/coach/assets/src/state/actions/exam.js

```diff
@@ -101,111 +148,80 @@ function showExamsPage(store, classId) {
+            exercises: content.exercises,
+            modalShown: false,
+          };
+
+          store.dispatch('SET_PAGE_STATE', pageState);
+          store.dispatch('CORE_SET_ERROR', null);
+          store.dispatch('CORE_SET_PAGE_LOADING', false);
+        },
+        error => {
+          CoreActions.handleError(store, error);
+        }
+      );
     },
     error => {
       CoreActions.handleError(store, error);
     }
   );
 }
 
-
-function _crumbState(ancestors) {
-  // skip the root node
-  return ancestors.slice(1).map(ancestor => ({
-    id: ancestor.pk,
-    title: ancestor.title,
-  }));
-}
-
-
-function _topicState(data) {
-  const state = {
-    id: data.pk,
-    title: data.title,
-    description: data.description,
-    breadcrumbs: _crumbState(data.ancestors),
-    next_content: data.next_content,
-  };
-  return state;
-}
-
-function _contentState(data) {
-  let progress;
-  if (!data.progress_fraction) {
-    progress = 0.0;
-  } else if (data.progress_fraction > 1.0) {
-    progress = 1.0;
-  } else {
-    progress = data.progress_fraction;
-  }
-  const state = {
-    id: data.pk,
-    title: data.title,
-    kind: data.kind,
-    description: data.description,
-    thumbnail: data.thumbnail,
-    available: data.available,
-    files: data.files,
-    progress,
-    content_id: data.content_id,
-    breadcrumbs: _crumbState(data.ancestors),
-    next_content: data.next_content,
-    author: data.author,
-    license: data.license,
-    license_owner: data.license_owner,
-  };
-  return state;
-}
-
-function _collectionState(data) {
-  const topics = data
-    .filter((item) => item.kind === CoreConstants.ContentNodeKinds.TOPIC)
-    .map((item) => _topicState(item));
-  const contents = data
-    .filter((item) => item.kind !== CoreConstants.ContentNodeKinds.TOPIC)
-    .map((item) => _contentState(item));
-  return { topics, contents };
-}
-
-function getChannelExercises(store, channelId, channelRootPk) {
-  const channelPayload = { channel_id: channelId };
-  const topicPromise = ContentNodeResource.getModel(channelRootPk, channelPayload).fetch();
-  const childrenPromise =
-    ContentNodeResource.getCollection(channelPayload, { parent: channelRootPk }).fetch();
-
-  ConditionalPromise.all([topicPromise, childrenPromise]).only(
-    CoreActions.samePageCheckGenerator(store),
-    ([topic, children]) => {
-      // store.dispatch('SET_TOPIC', topic);
-      console.log(_topicState(topic));
-      const collection = _collectionState(children);
-      console.log(collection.topics);
-      console.log(collection.contents);
-
-
-      store.dispatch('CORE_SET_PAGE_LOADING', false);
-      store.dispatch('CORE_SET_ERROR', null);
-    },
-    error => { CoreActions.handleApiError(store, error); }
-  );
+function createExam(store, classId, channelId, exercisesSelected, seed) {
+  store.dispatch('CORE_SET_PAGE_LOADING', true);
+  router.getInstance().push({ name: Constants.PageNames.EXAMS });
```

Comment:
This function is missing a lot of it's logic, so all those parameters will be used.